### PR TITLE
no ordering in maps

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**
Example failure: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/test-infra/29038/pull-test-infra-unit-test/1635754287025885184

/kind failing-test
/kind cleanup

this is just a quick fix

### CODE REVIEW

File: prow/config/tide_test.go [hunk 1]
```diff
@@ -1095,7 +1097,7 @@ func TestOrgRepoMatchMergeMethod(t *testing.T) {
 		test := test
 		t.Run(test.name, func(t *testing.T) {
 			actual := test.config.OrgRepoBranchMergeMethod(OrgRepo{Org: test.org, Repo: test.repo}, test.branch)
-			if actual != test.expected {
+			if actual != test.expected && actual != test.alternativeExpected {
```
Comment (aojea) [inline prow/config/tide_test.go:1100]: without knowing anything about these tests or code this looks like very aleatory behavior
why should it have alternatives modes and not be more determinisic?

Comment (SergeyKanzhelev) [reply prow/config/tide_test.go:1100]: yeah. I think adding ordering would be the right fix. Trying to deflake

**Comment by SergeyKanzhelev** [comment]
introduced in https://github.com/kubernetes/test-infra/pull/27962

@danilo-gemoli, @petr-muller, @jmguzik

File: prow/config/tide_test.go (hunk 1 above)
Comment (BenTheElder) [reply prow/config/tide_test.go:1100]: doesn't this mean the actual implementation behavior is non-deterministic? seems like we need to fix that.

Comment (SergeyKanzhelev) [reply prow/config/tide_test.go]: fixed properly now

**Review by jmguzik — APPROVED** [review]

**Comment by jmguzik** [comment]
@droslean could you approve the fix?

**Review by droslean — APPROVED** [review]

**Review by BenTheElder — COMMENTED** [review]
thanks!

**Comment by danilo-gemoli** [comment]
@SergeyKanzhelev good catch, thank you. It went unnoticed while I was developing the feature.
Even though I'm afraid we need to somehow preserve the ordering coming from the configuration rather than sorting it in place.
I'm going to open a PR in the next days.

**Pull request merged**